ksqldb-tools/src/test/java/io/confluent/ksql/tools/migrations/MigrationsTest.java

```diff
@@ -88,8 +96,44 @@ public static void classTearDown() {
   }
 
   @Test
-  public void dummy() {
-    // placeholder until additional functionality (beyond initialization) is added
+  public void testApply() throws IOException {
+    // Migration file
+    createMigrationFile(
+        1,
+        "foo",
+        MigrationsDirectoryUtil.getMigrationsDirFromConfigFile(configFilePath),
+        "CREATE STREAM FOO (A STRING) WITH (KAFKA_TOPIC='FOO', PARTITIONS=1, VALUE_FORMAT='DELIMITED');"
+    );
+    createMigrationFile(
+        2,
+        "bar",
+        MigrationsDirectoryUtil.getMigrationsDirFromConfigFile(configFilePath),
+        "CREATE STREAM BAR (A STRING) WITH (KAFKA_TOPIC='BAR', PARTITIONS=1, VALUE_FORMAT='DELIMITED');"
+    );
+    final int status = MIGRATIONS_CLI.parse("--config-file", configFilePath, "apply").run();
+    assertThat(status, is(0));
+
+    // verify FOO and BAR were registered
+    describeSource("FOO");
+    describeSource("BAR");
+
+    // verify current
+    final List<String> current = makeKsqlQuery("SELECT * FROM migration_schema_versions WHERE VERSION_KEY='CURRENT';");
```

Comment:
Feel free to defer to a follow-up PR, but we should probably switch these assertions (probably also the `describeSource()` assertions above and in the `initialize` tests) to `assertThatEventually()` to account for the async nature of these commands (especially for updating the metadata table's state store).